src/scene/immediate.js

```diff
@@ -266,8 +266,10 @@ Object.assign(pc.Application.prototype, function () {
         } else if (arg3) {
             // options passed in
             options = arg3;
-        } else {
-            // no arg3, use default options
+        }
+
+        // use default options
+        if (!options) {
```

Comment:
Would it be better to fix `_addLines` to handle an undefined options object? That way, you would avoid allocating a JS object on every call to this function.

Reply:
done, good idea.